### ISSUE TITLE
Markdown: coloring wrong after fenced code block

### ISSUE THREAD

**Issue by bpasero** [Microsoft/vscode#8259]
Refs: https://github.com/Microsoft/vscode/issues/7945

![image](https://cloud.githubusercontent.com/assets/900690/16409968/79f1aa4a-3d20-11e6-97e1-23afa1ed0b7e.png)

The markdown content is:

`````` markdown
# amdefine

A module that can be used to implement AMD's define() in Node. This allows you
to code to the AMD API and have the module work in node programs without
requiring those other programs to use AMD.

## Usage

**1)** Update your package.json to indicate amdefine as a dependency:

```javascript
    "dependencies": {
        "amdefine": ">=0.1.0"
    }
``````

Then run `npm install` to get amdefine into your project.

**2)** At the top of each module that uses define(), place this code:

``` javascript
if (typeof define !== 'function') { var define = require('amdefine')(module) }
```

**Only use these snippets** when loading amdefine. If you preserve the basic structure,
with the braces, it will be stripped out when using the [RequireJS optimizer](#optimizer).

You can add spaces, line breaks and even require amdefine with a local path, but
keep the rest of the structure to get the stripping behavior.

As you may know, because `if` statements in JavaScript don't have their own scope, the var
declaration in the above snippet is made whether the `if` expression is truthy or not. If
RequireJS is loaded then the declaration is superfluous because `define` is already already
declared in the same scope in RequireJS. Fortunately JavaScript handles multiple `var`
declarations of the same variable in the same scope gracefully.

```
```

**Comment by kieferrm** [comment]
I'm not yet sure what happens. But when I add a`\n` or `;` in front of the last '}' the tokenization is correct.

It seems the minimal case causing the wrong tokenization is:

![screen shot 2016-06-30 at 7 16 20 pm](https://cloud.githubusercontent.com/assets/4674940/16509991/2ce27328-3ef7-11e6-8e54-b731c307efd0.png)

**Comment by mjbvz** [comment]
Most cases should be fixed by #14626 . I've tested the cases in this issue, and they work fine:

![screen shot 2016-11-10 at 4 01 13 pm](https://cloud.githubusercontent.com/assets/12821956/20199386/ec42a0f6-a75e-11e6-9507-77aa8964a30a.png)


![screen shot 2016-11-10 at 4 01 59 pm](https://cloud.githubusercontent.com/assets/12821956/20199402/086d9330-a75f-11e6-8aea-bf498611ebda.png)
